R/derive_param_lasta.R

```diff
@@ -0,0 +1,358 @@
+#'   ```
+#'   pd_date <- date_source(
+#'     dataset_name = "adrs",
+#'     date = ADT,
+#'     filter = PARAMCD == PD
+#'   )
+#'   ```
+#'   and the actual response dataframe in the script is `myadrs`, `source_datasets
+#'   = list(adrs = myadrs)` should be specified.
+#'
+#'    *Required or Optional:* Optional
+#'
+#' @param subject_keys Columns to uniquely identify a subject
+#'
+#'   A list of symbols created using `admiral::vars()`.
+#'
+#'   *Default:* `admiral::vars(STUDYID, USUBJID)`
+#'
+#'   *Required or Optional:* Required
+#'
+#' @param set_values_to Columns to set
+#'
+#'   A named list returned by `vars()` defining the columns to be set for the
+#'   new parameter, e.g. `vars(PARAMCD = "LSTAC", PARAM = "Last Disease
+#'    Assessment Censored at First PD by Investigator")` is expected. The values
+#'    must be symbols, character strings, numeric values, or `NA`.
+#'
+#'    *Required or Optional:* Required
+#'
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++|
+#' @examples
+#'
+#' \dontrun{
+#'
+#' library(magrittr)
+#'
+#' adsl <- tibble::tribble(
+#'   ~USUBJID, ~TRTSDT,                      ~EOSDT,
+#'   "01",     lubridate::ymd("2020-12-06"), lubridate::ymd("2022-03-06"),
+#'   "02",     lubridate::ymd("2021-01-16"), lubridate::ymd("2022-02-03"),
+#'   "03",     lubridate::ymd("2021-01-09"), lubridate::ymd("2021-02-24"),
+#'   "04",     lubridate::ymd("2021-04-21"), lubridate::ymd("2021-09-15")
+#' ) %>%
+#'   dplyr::mutate(STUDYID = "a_study_id")
+#'
+#' adrs <- tibble::tribble(
+#'   ~USUBJID, ~PARAMCD, ~AVAL, ~AVALC, ~ASEQ, ~ADT,
+#'   "01", "RSP", NA, "Y", 1, lubridate::ymd("2021-04-08"),
+#'   "02", "RSP", NA, "N", 1, lubridate::ymd("2021-05-07"),
+#'   "03", "RSP", NA, "N", 1, NA,
+#'   "04", "RSP", NA, "N", 1, NA,
+#'   "01", "PD",  NA, "N", 1, NA,
+#'   "02", "PD",  NA, "Y", 1, lubridate::ymd("2021-05-07"),
+#'   "03", "PD",  NA, "N", 1, NA,
+#'   "04", "PD",  NA, "N", 1, NA,
+#'   "01", "OVR", 3, "SD", 1, lubridate::ymd("2021-03-07"),
+#'   "01", "OVR", 2, "PR", 1, lubridate::ymd("2021-04-08"),
+#'   "02", "OVR", 3, "SD", 1, lubridate::ymd("2021-03-07"),
+#'   "02", "OVR", NA, NA, 1, lubridate::ymd("2021-04-07"),
+#'   "02", "OVR", 6, "PD", 1, lubridate::ymd("2021-05-07"),
+#'   "03", "OVR", 3, "SD", 1, lubridate::ymd("2021-01-30"),
+#'   "03", "OVR", 3, "SD", 2, lubridate::ymd("2021-01-30"),
+#'   "04", "OVR", NA, "NE", 1, lubridate::ymd("2021-05-21"),
+#'   "04", "OVR", 5, "NON-PD", 1, lubridate::ymd("2021-06-30"),
+#'   "04", "OVR", NA, "NE", 1, lubridate::ymd("2021-07-24"),
+#'   "04", "OVR", NA, "ND", 1, lubridate::ymd("2021-09-30"),
+#' ) %>%
+#'   dplyr::mutate(STUDYID = "a_study_id")
+#'
+#' pd <- date_source(
+#'   dataset_name = adrs,
+#'   date         = ADT,
+#'   filter       = PARAMCD == 'PD' & AVALC == 'Y'
+#' )
+#'
+#' derive_param_lasta(
+#'   dataset,
+#'   filter_source = PARAMCD == "OVR" & ANL01FL == "Y",
+#'   source_pd = list(pd),
```

Comment:
this can just be "pd" - no need to use list() - that was my mistake in the issue

Reply:
Will update